api/user.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file
+
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/web-platform-tests/wpt.fyi/shared"
+)
+
+type loginResponse struct {
+	User  *shared.User `json:"user,omitempty"`
+	Error string       `json:"error,omitempty"`
+}
+
+func apiUserHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := shared.NewAppEngineContext(r)
+	aeAPI := shared.NewAppEngineAPI(ctx)
+	if !aeAPI.IsFeatureEnabled("githubLogin") {
+		http.Error(w, "Feature not enabled", http.StatusNotImplemented)
+		return
+	}
+
+	if cookie, err := r.Cookie("session"); err != nil || cookie == nil {
+		response := loginResponse{Error: "User is not logged in"}
+		marshalled, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write(marshalled)
+		return
+	}
```

Comment:
Isn't this basically the same as the check inside `GetUserFromCookie`?

Reply:
Well similar, originally I thought having this check would deliver a clear message  - whether a user is not logged in due to an empty session value vs internal serval error (corrupted cookie value, cookie expiry, etc). After a second thought, they should be treated the same because users have to log in again.